featuretools/entityset/entityset.py

```diff
@@ -730,7 +730,7 @@ def normalize_entity(self, base_entity_id, new_entity_id, index,
 
         if isinstance(make_time_index, str):
             base_time_index = make_time_index
-            new_entity_time_index = base_entity[make_time_index].id
```

Comment:
The only reason I can see for this is to check that the `make_time_index` exists in the base entity (if it doesn't then this would raise). If that should be checked here then I think it would be better to add an explicit assertion.

Reply:
Yes let's add an explicit assertion

Reply:
It looks like this is already checked when the entity is created:
https://github.com/Featuretools/featuretools/blob/4b6a5a4d38de94a64c08ad15b5b8cefa9954b6df/featuretools/entityset/entity.py#L574-L575
Is it worthwhile to add an earlier check?

Reply:
Let's split this off to a separate issue to be dealt with alter.  Looks like there are two situations where we could improve the error message:
* Specifying an existing column for `make_time_index` but not adding that column to `additional_variables` or `copy_variables`
* Setting `make_time_index` to a column that doesn't exist in the base frame

Reply:
👍 I can open an issue.